utils/default.nix

```
@@ -6,6 +6,10 @@ rustPlatform.buildRustPackage {
   src = ./.;
   cargoLock.lockFile = ./Cargo.lock;
 
+  preBuild = ''
+    export HOME=$(mktemp -d)
```

```suggestion
    export HOME=${NIX_BUILD_TOP}/.home
    mkdir -p $HOME
```

It's probably better to put the HOME directory into the build path so that it may be cleaned up by nix with the rest of the build files. At least that's what I've always done in similar situations